### PR TITLE
Rekey geniza secrets with secondary vault key and setup deploy with limited access

### PR DESCRIPTION
**Pull request by rlskoeser**
to address https://github.com/princeton-cdh/geniza/issues/1051

changes in this update:
- a new geniza ansible vault key has been added to CDH LastPass
- encrypted variables and files for geniza have been rekeyed with the new geniza vault key
- two shell scripts have been added to this repo to wrap around lpass cli to retrieve the appropriate vault key (doing this means we don't need to install the `lastpass-ansible` gem)
- `ansible.cfg` has been updated with default identity list for both the default key and the new geniza key
- unnecessarily vaulted qa variables have been moved to regular variables
- `all/vault.yml` variables has been moved to a vault var file in a new setup role, which is loaded as a dependency for other roles to conditionally loaded if setup tag is specified and the geniza deploy only environment variable is not set.
- update readme with basic lastpass info and setup instructions
- add section to readme about setup for running geniza playbooks with limited access

When I test with `ansible-vault view` and the default credentials, I can successfully decrypt and view vault files with either key. If I override the vault ids with an environment variable `setenv ANSIBLE_VAULT_IDENTITY_LIST geniza@bin/lpass_geniza.sh`, then I can decrypt the geniza vaulted files but decryption of other files fails.

I'm now able to run geniza qa playbook successfully with the geniza vault id only; when I try to run it with setup tasks enabled, it errors immediately. When I run with both geniza and default vault ids, I can run the playbook with setup tasks enabled.

Update 2023-01-12: had a working session with Ben (contractor) to get him set up; we worked from this branch and confirmed that he was able to run deploy scripts successfully. I updated from `main` and pulled in the replicate playbook, and confirmed that with a few minor tweaks (at least one of which I've made in the other branch), he was able to successfully replicate from production to qa.  I should improve the setup documentation, though — some of it is unclear and other parts are out of date.

### CODE REVIEW

File: bin/lpass_default.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+# use lastpass cli to get the default cdh-ansible vault password
+
+lpass show --password Ansible_Vault/cdh-ansible
```
Comment (quadrismegistus) [inline bin/lpass_default.sh:5]: This is nice!

**Review by quadrismegistus — APPROVED** [review]
Looks great! Thanks for doing all of that. Tough nitty gritty work but will help us a lot going forward.

File: roles/setup/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Requirements
+------------
+
+None
+
+Role Variables
+--------------
+
+Checks whether the `geniza_deploy_only` variable is defined to determine
```
Comment (acozine) [inline roles/setup/README.md:15]: Could we do all this in the `geniza_setup` role? Why is this necessary for testing the postgresql role?

**Review by acozine — COMMENTED** [review]
Impressive work. I've done a first review (mostly catching nitpicky grammar things), but I need to think more about how this integrates into the overall repo. I wonder if there's a way we could streamline this a bit.

**Comment by acozine** [comment]
This PR has uncovered some linting issues. In `roles/passenger`:
```
./handlers/main.yml
  Error: 16:1 [empty-lines] too many blank lines
```
In `roles/prosody_setup`:
```
./tasks/extract_marc.yml
  Error: 24:81 [line-length] line too long (144 > 80 characters)

./molecule/default/verify.yml
  Error: 9:3 [indentation] wrong indentation: expected at least 3
```

File: roles/setup/README.md (hunk 2 above)
Comment (acozine) [reply roles/setup/README.md:15]: Oh, never mind. I am confusing this repo with the princeton_ansible repo . . .

**Comment by rlskoeser** [comment]
> This PR has uncovered some linting issues. In `roles/passenger`:
> 
> ```
> ./handlers/main.yml
>   Error: 16:1 [empty-lines] too many blank lines
> ```
> 
> In `roles/prosody_setup`:
> 
> ```
> ./tasks/extract_marc.yml
>   Error: 24:81 [line-length] line too long (144 > 80 characters)
> 
> ./molecule/default/verify.yml
>   Error: 9:3 [indentation] wrong indentation: expected at least 3
> ```

Thank you for telling me about these — at some point my molecule tests broke (due to some change external to my ansible repo) and I haven't had time or energy to figure out why and fix; and unfortunately I've kind of let linting and testing slip since then
